src/main/java/org/sonarlint/intellij/ui/AbstractIssuesPanel.java

```diff
@@ -60,6 +89,67 @@ protected JComponent createSplitter(JComponent c1, JComponent c2, String proport
     return splitter;
   }
 
+  protected void flowsTreeSelectionChanged() {
+    LocationNode[] selectedNodes = flowsTree.getSelectedNodes(LocationNode.class, null);
+    if (selectedNodes.length > 0) {
+      LocationNode node = selectedNodes[0];
+      highlighting.highlightFlowsWithHighlightersUtil(node.rangeMarker(), node.message(), Collections.emptyList());
+    }
+  }
+
+  protected void issueTreeSelectionChanged() {
+    IssueNode[] selectedNodes = tree.getSelectedNodes(IssueNode.class, null);
+    if (selectedNodes.length > 0) {
+      LiveIssue issue = selectedNodes[0].issue();
+      rulePanel.setRuleKey(issue);
+      if (issue.getRange() != null) {
+        highlighting.highlightFlowsWithHighlightersUtil(issue.getRange(), issue.getMessage(), issue.flows());
+      }
+      flowsTree.getEmptyText().setText("Selected issue doesn't have flows");
+      flowsTreeBuilder.setFlows(issue.flows(), issue.getRange(), issue.getMessage());
+      flowsTree.expandAll();
+    } else {
+      flowsTreeBuilder.clearFlows();
+      flowsTree.getEmptyText().setText("No issue selected");
+      rulePanel.setRuleKey(null);
+      highlighting.removeHighlightingFlows();
+    }
+  }
+
+  protected void addToolbar() {
+    ActionGroup mainActionGroup = (ActionGroup) ActionManager.getInstance().getAction(getToolbarGroupId());
+    mainToolbar = ActionManager.getInstance().createActionToolbar(ID, mainActionGroup, false);
+    mainToolbar.setTargetComponent(this);
+    Box toolBarBox = Box.createHorizontalBox();
+    toolBarBox.add(mainToolbar.getComponent());
+
+    super.setToolbar(toolBarBox);
+    mainToolbar.getComponent().setVisible(true);
+  }
+
+  private void createFlowsTree() {
+    flowsTreeBuilder = new FlowsTreeModelBuilder();
+    DefaultTreeModel model = flowsTreeBuilder.createModel();
+    flowsTree = new FlowsTree(project, model);
+    flowsTreeBuilder.clearFlows();
+    flowsTree.getEmptyText().setText("No issue selected");
+    flowsTree.addTreeSelectionListener(e -> flowsTreeSelectionChanged());
+  }
+
+  private void createIssuesTree() {
+    treeBuilder = new IssueTreeModelBuilder();
+    DefaultTreeModel model = treeBuilder.createModel();
+    tree = new IssueTree(project, model);
+    tree.addTreeSelectionListener(e -> issueTreeSelectionChanged());
+    tree.addKeyListener(new KeyAdapter() {
+      public void keyPressed(KeyEvent e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1161)